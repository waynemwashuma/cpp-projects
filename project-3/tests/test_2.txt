x, y, z;

hello {
  public: a, b;
  private: c, d;

  x = a;
  c = y;
  w1 {
    private: a, b;
   
    a = x;
    y = c;
    b = d;
  }
  b = z;
  y = d;
  w2 {
    public: a, b;

    x = a;
    c = y;
    d = b;
  }
  d = b;
  y = z;
  w3 {
    x = a;
    y = c;
    d = b;
  }
  x = y;
  b = z;
}