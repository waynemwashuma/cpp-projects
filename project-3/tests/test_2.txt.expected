::x = hello.a
hello.c = ::y
w1.a = ::x
::y = ?.c
w1.b = ?.d
hello.b = ::z
::y = hello.d
::x = w2.a
?.c = ::y
?.d = w2.b
hello.d = hello.b
::y = ::z
::x = hello.a
::y = ?.c
?.d = hello.b
::x = ::y
hello.b = ::z
